lib/go/thrift/uuid.go

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package thrift
+
+import (
+	"encoding/hex"
+	"errors"
+)
+
+// most of the code is copied from https://github.com/google/uuid in order to reduce dependency on
+// external libraries
```

Comment:
if we are copying code then the license notification would need to be different to reflect the code we copied.

I really did not intend to copy any code. the majority of the complexity in uuid third party libraries are in the generating part, which we don't need in thrift, and that's why I prefer to **reimplement** the parts we need in thrift over depending on a third party library. do you need this merged in urgently? do you mind me taking this ticket later this week instead?

Reply:
@fishy no i'm not in a hurry so please feel free to take your time and grab this pr and make changes to it. thanks!